### PR TITLE
Misc improvements.

### PR DESCRIPTION
**Pull request by aomarks**
See commits.

### CODE REVIEW

File: src/closure-types.ts [hunk 1]
```diff
@@ -96,20 +104,42 @@ function nullableByDefault(node: doctrine.Type): boolean {
...
+}
+
 function serializeArray(node: doctrine.type.TypeApplication): string {
   if (node.applications.length !== 1) {
     console.error('Invalid array expression.');
     return '';
   }
-  return serialize(node.applications[0]) + '[]';
+  const arg = node.applications[0];
+  return serialize(arg, true) + '[]';
```
Comment (rictic) [inline src/closure-types.ts]: Consider just doing `Array<Foo>` in all cases. It's equivalent and scales better to complicated types.

File: src/test/goldens/polymer/expected.d.ts [hunk 2]
```diff
@@ -819,7 +819,7 @@ declare namespace Polymer {
      * function should match the sort function passed to `Array.sort`.
      * Using a sort function has no effect on the underlying `items` array.
      */
-    sort: Function|null;
```
Comment (rictic) [inline src/test/goldens/polymer/expected.d.ts]: Function is legal TypeScript, and it has the same semantics AFAIK as Closure's, why not just keep it that way?

**Review by rictic — APPROVED** [review]
Neither of the above are blocking though.

File: src/closure-types.ts (hunk 1 above)
Comment (aomarks) [reply src/closure-types.ts]: I like it, simplifies parenthesis handling too. Done.

**Pull request merged**